ddtelemetry/src/worker/mod.rs

```diff
@@ -296,7 +293,9 @@ impl TelemetryWorker {
                     self.log_err(&e);
                 }
                 self.data.started = false;
-                self.deadlines.clear_pending();
+                if !self.config.restartable {
```

Comment:
Wouldn't it be enough to only include `self.data.started = false;` inside the if statement as well, and leave the exit early checks?

Reply:
I guess it depends. Do we want Start, Stop, Stop to generate two stops? Because that's what PHP ends up generating. The second stop is a noop, but if I moved the assignment `self.data.started = false` under the condition `if !self.config.restartable`, then the stops would be effective

Reply:
If the exit early checks are still in the code, then how would the second `Stop` be effective?

Reply:
From what I understand, what you're proposing is that I reset `started = false` only if `!restartable`. In that case `started` would stay true forever once there is a start. So the early check

```
 if !self.data.started {
   return BREAK;
 }
```

would never be hit.

Reply:
Yes, you're right. I was confusing the early checks.

There is something in the logic that feels a bit broken. The `FlushData` is also protected by this `!self.data.started` check. Should it work after a _restartable_ stop?

The things that `Stop` does, should they happen when the first request ends, or when the worker stops?